lib/inspec/base_cli.rb

```diff
@@ -71,24 +73,70 @@ def self.exec_options
         desc: 'Write out a lockfile based on this execution (unless one already exists)'
       option :backend_cache, type: :boolean,
         desc: 'Allow caching for backend command output.'
+      option :show_progress, type: :boolean,
+        desc: 'Show progress while executing tests.'
     end
 
     def self.default_options
       {
         exec: {
+          'reporter' => {
+            'cli' => nil,
+          },
+          'show_progress' => false,
           'color' => true,
           'create_lockfile' => true,
           'backend_cache' => false,
         },
       }
     end
 
+    def self.clean_reporters(opts)
+      # Parse report options from CLI
+      if opts['reporter'].is_a?(Array)
+        stdout = 0
+        reports = {}
+        opts['reporter'].each do |report|
+          k, v = report.split(':')
+          v = nil if v == '-'
+          reports[k] = v
+          stdout += 1 if v.nil?
+        end
+
+        raise ArgumentError, 'The option --reporter can only have a single report outputting to stdout.' if stdout > 1
+
+        opts['reporter'] = reports
+      end
+
+      # Catch anyone using the legacy --format option
+      unless opts['format'].nil?
```

Comment:
To ensure backward compatibility (because of our desire to get this in to 1.x), `--format` and `--reporter` have to be able to co-exist. And a big deliverable of this change is that multiple reporters/formatters can be used simultaneously - a HUGE persistent request from the community. I think it's only unintuitive because you're reading the code, but a everyday InSpec user shouldn't even notice unless they become aware of the new feature and start using it.

Totally agree with the deprecation warning though. That could be added in a 2.x version though because it's way too late to deprecate this for 2.0.